modules/peridynamics/src/userobjects/SingularShapeTensorEliminatorUserObjectPD.C

```diff
@@ -0,0 +1,163 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SingularShapeTensorEliminatorUserObjectPD.h"
+#include "AuxiliarySystem.h"
+
+registerMooseObject("PeridynamicsApp", SingularShapeTensorEliminatorUserObjectPD);
+
+InputParameters
+SingularShapeTensorEliminatorUserObjectPD::validParams()
+{
+  InputParameters params = GeneralUserObjectBasePD::validParams();
+  params.addClassDescription("UserObject to eliminate the existance of singular shape tensor");
+
+  MooseEnum formulation_option("CONVENTION BONDHORIZON");
+  params.addRequiredParam<MooseEnum>(
+      "formulation", formulation_option, "Which form of shape tensor used in the formulation");
+
+  return params;
+}
+
+SingularShapeTensorEliminatorUserObjectPD::SingularShapeTensorEliminatorUserObjectPD(
+    const InputParameters & parameters)
+  : GeneralUserObjectBasePD(parameters), _formulation(getParam<MooseEnum>("formulation"))
+{
+}
+
+void
+SingularShapeTensorEliminatorUserObjectPD::initialize()
+{
+  _aux.solution().close();
+}
+
+void
+SingularShapeTensorEliminatorUserObjectPD::execute()
+{
+  bool converged = false;
+
+  // Loop through the active local elements to check the shape tensor singularity
+  auto first_elem = _mesh.getMesh().active_local_elements_begin();
+  auto last_elem = _mesh.getMesh().active_local_elements_end();
+
+  unsigned int singularity_count, loop_count = 0;
+  bool should_print = true; // for printing purpose only
```

Comment:
The print statement was intended to be printed only once. If this parameter was replaced by _if (singularity_count == 1)_ , the print statement will be printed at the beginning of each iteration, since _singularity_count_ will be reinitialized to be zero at the beginning of each iteration.